lib/topological_inventory/openshift/operations/worker.rb

```diff
@@ -82,6 +76,68 @@ def update_task(task_id, state:, status:, context:)
           api_client.update_task(task_id, task)
         end
 
+        def svc_instance_context_with_url(service_offering, service_plan, service_instance)
+          context = {
+            :service_instance => {
+              :source_id  => service_plan.source_id,
+              :source_ref => service_instance.spec&.externalID
+            }
+          }
+
+          if provisioning_status(service_instance) == "ok"
+            context[:service_instance][:url] = svc_instance_url(service_offering, service_instance)
+          end
+
+          context
+        end
+
+        def svc_instance_url(service_offering, service_instance)
+          svc_instance = svc_instance_by_source_ref(service_offering.source_id,
+                                                    service_instance.spec&.externalID)
+
+          rest_api_path = '/service_instances/{id}'.sub('{' + 'id' + '}', svc_instance&.id.to_s)
+          api_client.api_client.build_request(:GET, rest_api_path).url
+        end
+
+        def provisioning_status(service_instance)
+          reason = service_instance.status.conditions.first&.reason
+          reason == "ProvisionedSuccessfully" ? "ok" : "error"
+        end
+
+        # Current API client doesn't support source_id and source_ref filtering
+        # This is modified version of api_client.list_service_instances
```

Comment:
@slemrmartin how about using `/sources/#{source_id}/service_instances?source_ref=#{source_ref}` ?

Reply:
@agrare it's definitely possible, but then it should be written all the error handling and configuration code which is written in API too, so I doubt if it will be more readable

Reply:
:+1: yeah exactly